File: BEPiD/BepidAulas/AutoLayout/AutoLayout/AutoLayout/ViewController.m
//
//  ViewController.m
//  AutoLayout
//
//  Created by Danilo Barros Mendes on 6/29/15.
//  Copyright (c) 2015 Danilo Barros Mendes. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)alterar:(UIButton *)sender {
    
    if(self.vermelhoHeight.constant != 50){
        self.verdeHeight.constant = 50.0;
        [self.vermelhoHeight setConstant:50];
    }else{
        self.verdeHeight.constant = 155;
        [self.vermelhoHeight setConstant:155];
    }
}

@end

File: BEPiD/BepidAulas/aula24/CameraSwift/CameraSwift/ViewController.swift
//
//  ViewController.swift
//  CameraSwift
//
//  Created by Danilo Barros Mendes on 5/18/15.
//  Copyright (c) 2015 Danilo Barros Mendes. All rights reserved.
//

import UIKit

class ViewController: UIViewController,
        UINavigationControllerDelegate,
        UIImagePickerControllerDelegate,
        UITextFieldDelegate{

    
    @IBOutlet weak var imageView: UIImageView!
    
    @IBOutlet weak var camera: UIBarButtonItem!
    
    @IBOutlet weak var textField: UITextField!
    
    
    
    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

        //MARK: - TextField
    
    func textFieldShouldReturn(textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        return true
    }
    
    @IBAction func backgroundTapped(sender: AnyObject) {
        view.endEditing(true)
    }
    
    //MARK: - Camera
    
    @IBAction func takePicture(sender: UIBarButtonItem) {
        
        let imagePicker = UIImagePickerController();
        
        imagePicker.allowsEditing = true
        
        
        //Se a camera existir
        
        if UIImagePickerController.isSourceTypeAvailable(.Camera){
            
            imagePicker.sourceType = .Camera
            
        }else{
            imagePicker.sourceType = .PhotoLibrary;
        }
        
        imagePicker.delegate = self
        
        presentViewController(imagePicker, animated: true, completion: nil)
        
    }
    
    @IBAction func deleteImage(sender: UIBarButtonItem) {
        
        
        imageView.image = nil
        
    }
    
    
    func imagePickerController(picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [NSObject : AnyObject]) {
        
        let image = info[UIImagePickerControllerEditedImage] as! UIImage
        
        imageView.image = image
        
        dismissViewControllerAnimated(true, completion: nil)
        
    }
    
    /*
    // MARK: - Navigation

    // In a storyboard-based application, you will often want to do a little preparation before navigation
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        // Get the new view controller using segue.destinationViewController.
        // Pass the selected object to the new view controller.
    }
    */

}
